maestro-ios/src/main/java/ios/xcrun/Simctl.kt

```diff
@@ -106,4 +107,63 @@ object Simctl {
 
         reboot(deviceId)
     }
-}
\ No newline at end of file
+
+    private enum class Permission {
+        ALLOW,
+        REVOKE,
+        RESET
+    }
+
+    private val allPermissions = listOf(
+        "calendar",
+        "camera",
+        "contacts",
+        "faceid",
+        "health",
+        "homekit",
+        "location",
+        "medialibrary",
+        "microphone",
+        "motion",
+        "notifications",
+        "photos",
+        "reminders",
+        "siri",
+        "speech",
+        "userTracking",
+    )
+
+    private fun setPermissionsTo(deviceId: String, bundleId: String, permissions: Map<String, Permission>) {
+        val setPermissions = permissions
+            .mapValues { (_, value) ->
+                when (value) {
+                    Permission.ALLOW -> "YES"
+                    Permission.REVOKE -> "NO"
+                    Permission.RESET -> "unset"
+                }
+            }
+            .map { (key, value) -> "$key=$value" }
+            .joinToString { "," }
+
+        runCommand(
+            listOf(
+                "applesimutils",
```

Comment:
Can we somehow auto-install `applesimutils` if they are not installed automatically?

Reply:
Only if the user already has homebrew installed. I think it wouldn't be acceptable to install homebrew (which takes >5 minutes) without checking in with the user.

Reply:
One trick would be to package the binary with maestro, or download it directly from homebrew kegs.

Reply:
Is applesimutils installed by default with Xcode?

Reply:
No, it is a 3rd party tool

Reply:
Hm yeah this poses a pretty serious usability issue. What does applesimutils do under the hood can we mimic what it's doing manually?

Reply:
I think the logic in applesimutils is rather straightforward and could be copied over (i.e. reading from SQLite DB), though it will take a bit of time to do

Reply:
Sounds like the right approach to me